gstreamer-plugin/git-patch/0001-svthevcenc-Add-svthevc-encoder-element.patch

```diff
@@ -0,0 +1,2059 @@
++      break;
++  }
++
++  return res;
++}
++
++static GstCaps *
++gst_svthevc_enc_sink_getcaps (GstVideoEncoder * enc, GstCaps * filter)
++{
++  GstCaps *supported_incaps;
++  GstCaps *ret;
++
++  supported_incaps = gst_svthevc_enc_get_supported_input_caps ();
++
++  ret = gst_video_encoder_proxy_getcaps (enc, supported_incaps, filter);
++  if (supported_incaps)
++    gst_caps_unref (supported_incaps);
++  return ret;
++}
++
++static void
++gst_svthevc_enc_class_init (GstSvtHevcEncClass * klass)
++{
++  GObjectClass *gobject_class;
++  GstElementClass *element_class;
++  GstVideoEncoderClass *gstencoder_class;
++
++  gobject_class = G_OBJECT_CLASS (klass);
++  element_class = GST_ELEMENT_CLASS (klass);
++  gstencoder_class = GST_VIDEO_ENCODER_CLASS (klass);
++
++  gobject_class->set_property = gst_svthevc_enc_set_property;
++  gobject_class->get_property = gst_svthevc_enc_get_property;
++  gobject_class->finalize = gst_svthevc_enc_finalize;
++
++  gstencoder_class->set_format = GST_DEBUG_FUNCPTR (gst_svthevc_enc_set_format);
++  gstencoder_class->handle_frame =
++      GST_DEBUG_FUNCPTR (gst_svthevc_enc_handle_frame);
++  gstencoder_class->start = GST_DEBUG_FUNCPTR (gst_svthevc_enc_start);
++  gstencoder_class->stop = GST_DEBUG_FUNCPTR (gst_svthevc_enc_stop);
++  gstencoder_class->flush = GST_DEBUG_FUNCPTR (gst_svthevc_enc_flush);
++  gstencoder_class->finish = GST_DEBUG_FUNCPTR (gst_svthevc_enc_finish);
++  gstencoder_class->getcaps = GST_DEBUG_FUNCPTR (gst_svthevc_enc_sink_getcaps);
++  gstencoder_class->sink_query = GST_DEBUG_FUNCPTR (gst_svthevc_enc_sink_query);
++  gstencoder_class->propose_allocation =
++      GST_DEBUG_FUNCPTR (gst_svthevc_enc_propose_allocation);
++
++  g_object_class_install_property (gobject_class, PROP_INSERT_VUI,
++      g_param_spec_boolean ("insert-vui", "Insert VUI",
++          "Insert VUI NAL in stream",
++          PROP_INSERT_VUI_DEFAULT, G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_AUD,
++      g_param_spec_boolean ("aud", "AUD",
++          "Use AU (Access Unit) delimiter", PROP_AUD_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_HIERARCHICAL_LEVEL,
++      g_param_spec_uint ("hierarchical-level", "Hierarchical Level",
++          "Hierarchical prediction levels setting",
++          0, 3, PROP_HIERARCHICAL_LEVEL_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_LOOKAHEAD_DISTANCE,
++      g_param_spec_uint ("lookahead-depth", "Lookahead Depth",
++          "Look ahead distance",
++          0, 250, PROP_LOOKAHEAD_DISTANCE_DEFAULT,
++          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
++
++  g_object_class_install_property (gobject_class, PROP_ENCODER_MODE,
++      g_param_spec_uint ("encoder-mode", "Encoder Mode",
```

Comment:
x264 use "speed-preset", gst-svt use "speed" for encoder-mode, so how about we use "speed" here?

Reply:
I think too that would be better :)